charmorigin.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+)
+
+// CharmOriginArgs is an argument struct used to add information about the
+// tools the agent is using to a Machine.
+type CharmOriginArgs struct {
+	Source string
+}
+
+func newCharmOrigin(args CharmOriginArgs) *charmOrigin {
+	return &charmOrigin{
+		Version_: 1,
+		Source_:  args.Source,
+	}
+}
+
+// Keeping the charmOrigin with the machine code, because we hope
+// that one day we will succeed in merging the unit agents with the
+// machine agents.
+type charmOrigin struct {
+	Version_ int    `yaml:"version"`
```

Comment:
Do we want to add a channel value as well?  With charm hub, version may not be enough to determine exact charm.